clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -207,6 +208,26 @@ public void testFetchNormal() {
         }
     }
 
+    @Test
+    public void testFetcherCloseClosesFetchSessionsInBroker() throws Exception {
+        subscriptions.assignFromUser(singleton(tp0));
+        subscriptions.seek(tp0, 0);
+        client.prepareResponse(fullFetchResponse(tp0, this.records, Errors.NONE, 100L, 0));
+        fetcher.sendFetches();
+        consumerClient.poll(0);
+        assertEquals(0, consumerClient.pendingRequestCount(node));
+        this.fetcher.close(); // should send request to close the session
+
+        assertEquals(1, consumerClient.pendingRequestCount(node));
+        assertEquals(0, client.inFlightRequestCount());
+        consumerClient.poll(0);
+        assertEquals(1, client.inFlightRequestCount());
+        assertEquals(1, consumerClient.pendingRequestCount(node));
+        FetchRequest sessionCloseReq = (FetchRequest) client.requests().peek().requestBuilder().build();
+        assertEquals(FetchMetadata.FINAL_EPOCH, sessionCloseReq.metadata().epoch());  // final epoch indicates we want to close the session
```

Comment:
Should we also check that there are no partitions contained in the request?

Reply:
Also we should match the fetch session with the earlier one

Reply:
Absolutely. Done